docs/index.mdx

```diff
@@ -33,20 +33,91 @@ Or load from a CDN:
 
 ## Example Usage
 The first argument is data. This is all that is required.
+
+### JSON to CSV
+```.jsx
+  <button onClick={() => csvDownload(jsonData)}>
+    Download CSV Data
+  </button>
+```
+
+### CSV to JSON
 ```.jsx
-  <button onClick={() => csvDownload(mockData)}>
-    Download Data
+  <button onClick={() => jsonDownload(csvData)}>
+    Download JSON Data
   </button>
 ```
 
-## Arguments
+## API
+
+### csvDownload
+Singular function to convert and download JSON to CSV data. This is shown in an example above.
+
+| #   | Argument  | Type     | Requirement | Default      | Description                |
+| --- | --------- | -------- | ----------- | ------------ | -------------------------- |
+| 1   | data      | `object` | `required`  | `null`       | object or array of objects |
+| 2   | filename  | `string` | `optional`  | "export.csv" | The complete filename      |
+| 3   | delimiter | `string` | `optional`  | ","          | field separator            |
+
+#### returns
+```
+VOID
+```
+
+### jsonDownload
+Singular function to convert and download CSV to JSON data. This is shown in an example above.
+
+| #   | Argument  | Type     | Requirement | Default        | Description                 |
+| --- | --------- | -------- | ----------- | ------------   | --------------------------  |
+| 1   | data      | `string` | `required`  | `null`         | string                      |
+| 2   | filename  | `string` | `optional`  | "export.json"  | The complete filename       |
+| 3   | delimiter | `string` | `optional`  | ","            | field separator             |
+
+#### returns
+```
+VOID
+```
+
+### download({filename, object, error})
+Download any converted data as long as it can be converted into a Blob object.
+
+| #   | Argument  | Type     | Requirement | Description                      |
+| --- | --------- | -------- | ----------- | --------------------------       |
+| 1   | filename  | `string` | `required`  | The filename for the download    |
+| 2   | object    | `any` | `required`     | The 'any' object to be exported |                      |
+| 3   | error     | `string` | `optional`  | Any errors to be reported. This will result in the download being stopped and a new error thrown. This option is used by the csvDownload and jsonDownload functions.            |
+#### returns
+```
+VOID
+```
+
+### csvToJson()
+Convert CSV data into JSON data.
+
+| #   | Argument  | Type     | Requirement | Default        | Description                 |
+| --- | --------- | -------- | ----------- | ------------   | --------------------------  |
+| 1   | data      | `string` | `required`  | `null`         | string                      |
+| 2   | filename  | `string` | `optional`  | "export.json"  | The complete filename       |
+| 3   | delimiter | `string` | `optional`  | ","            | field separator             |
```

Comment:
Four of these functions have the same arguments. The docs can be simplified to reflect that.